### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by h3xduck**
Fixed a bug where numbers with more than one decimal point where accepted, such as 9.2.3.
Rlly liked your project tho

### CODE REVIEW

File: calculator.py [hunk 1]
```diff
@@ -58,17 +58,24 @@ def checkOperators(expList):
...
 def checkDecimalPoint(expList):
 	expression = expList[0]
 	index = 0
-	if(expression[0]=="." or expression[-1]=="."): return False
-	for char in expression: #If there
-		if((expression[index]==".") and (index != 0) and (index != len(expression)-1) and (not(expression[index-1] in "0123456789" and expression[index+1] in "0123456789"))):
-			return False
+	validPointApperance="false";
+	#We make sure that, once received a point, we do not receive more on the same number
+	for char in expression:
```
Comment (MIBbrandon) [inline calculator.py:70]: It doesn't catch when the decimal point is at the beginning or at the end of an expression, nor when it is right before or after an operator.

File: calculator.py [hunk 2]
```diff
@@ -58,17 +58,24 @@ def checkOperators(expList):
...
+		if(expression[index]=="."):
+			if(validPointApperance=="false"):	
+				validPointApperance="true"
+			else:
+				return False
+		elif (expression[index] in "+-/*()"):
+			validPointApperance="false"		
+		index-=-1
```
Comment (MIBbrandon) [inline calculator.py:78]: Why not just index += 1?

File: calculator.py (hunk 1 above)
Comment (h3xduck) [reply calculator.py:70]: Neither does the original version if the number is not isolated, such as 3.+2, or if there exist an operator before the point, such as -.3+2. That's the reason I thought it was expected behavior.
A solution is to merge the new code without deleting the old function, getting rid of the bug I solved but keeping the one described above. I'll update the files and pull if it's OK for you

File: calculator.py (hunk 2 above)
Comment (h3xduck) [reply calculator.py:78]: Because.
https://i.redd.it/5gk732q3w0241.png

**Comment by h3xduck** [comment]
Issue solved

**Pull request closed without merging**